### ISSUE TITLE
OSX Yosemite install

### ISSUE THREAD

**Issue by JMDelhGit** [Project-OSRM/osrm-backend#2928]
Hi,

I'm trying to install the OSRM-backend locally on a iMac under OS X Yosemite 10.10.5

I have installed the brew package first.
I'm now following the instructions to build the library osrm-backend:

mkdir -p build
cd build 
  => ok
cmake .. -DCMAKE_BUILD_TYPE=Release
  => ok
cmake --build .
  => Error
I received different errors on type data that are not available in "cmath" include file:
The first error came at:
" Scanning dependencies of target STORAGE
[  0%] Building CXX object CMakeFiles/STORAGE.dir/src/storage/storage.cpp.o
In file included from /Users/JMDelhaze/osrm-backend/src/storage/storage.cpp:3:
In file included from /Users/JMDelhaze/osrm-backend/include/extractor/compressed_edge_container.hpp:6:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/unordered_map:350:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__hash_table:19:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/cmath:1337:9: error: no member named 'llrint' in the global namespace; did you mean 'lrint'?
using ::llrint;"

This error was followed by other having the same problem with "llrint" type  
The same happens with "llround", "lround", "lroundf", "llroundf" , "lroundl", "llroundl" types/functions also

The build stop with these errors.

Usually using Xcode, I don't know where to go from here. 
Could you help with this
Thank you in advance

**Comment by daniel-j-h** [comment]
Looks like a compiler/stdlib issue coming from

    #include <unordered_map>

can you try to put this and `int main() {}` into a file called `test.cc` and try to compile it e.g. with

    clang++ -Wall -Wextra -pedantic -std=c++1y test.cc



cc @springmeyer for OSX support.

**Comment by JMDelhGit** [comment]
Thank you for quick reaction.

Here is the test I just did:

//test.cc start
```
/* osrm compile test
test.cc
created 20160916
*/
#include <unordered_map>
#include <iostream>
int main()
{
    std::cout << "Hello, world!\n";
}

```//test.cc end

I then compiled following your template 
`clang++ -Wall -Wextra -pedantic -std=c++1y test.cc
`
=> No error

**Comment by JMDelhGit** [comment]
Hello,

I retried on another machine and ... it was ok!

Digging into the issue on the first machine, I find out that I installed a few years ago a CMake package and LLVM compilers which created includes items in the root /usr directory.

When preparing the build files, brew cmake did look at  /usr/Include instead of looking at Xcode application DSK. These  'llrint', ... types did not exist in the past.

I fixed the root  /usr items and recompiled the package 
=> OK

Problem is solved.

Thank you for the help
Jean Marie
